django/db/backends/sqlite3/schema.py

```diff
@@ -358,11 +358,22 @@ def _alter_field(self, model, old_field, new_field, old_type, new_type,
             return self.execute(self._rename_field_sql(model._meta.db_table, old_field, new_field, new_type))
         # Alter by remaking table
         self._remake_table(model, alter_field=(old_field, new_field))
-        # Rebuild tables with FKs pointing to this field if the PK type changed.
-        if old_field.primary_key and new_field.primary_key and old_type != new_type:
-            for rel in new_field.model._meta.related_objects:
-                if not rel.many_to_many:
-                    self._remake_table(rel.related_model)
+        # Rebuild tables with FKs pointing to this field was and is still
+        # referenceable.
+        if old_field.unique and new_field.unique and old_type != new_type:
```

Comment:
Only checking `new_field.unique` is sufficient here, it's really just an optimization to avoid the loop below really.

```suggestion
        if new_field.unique and old_type != new_type:
```